Sources/Foundation/NSLock.swift

```diff
@@ -49,7 +49,8 @@ private typealias _RecursiveMutexPointer = UnsafeMutablePointer<pthread_mutex_t>
 private typealias _ConditionVariablePointer = UnsafeMutablePointer<pthread_cond_t>
 #endif
 
-open class NSLock: NSObject, NSLocking {
+// fix for: https://github.com/apple/swift-corelibs-foundation/issues/4941
+open class NSLock: NSObject, NSLocking, @unchecked Sendable {
```

Comment:
You should look at amending `NSRecursiveLock` and `NSConditionLock` too.

Reply:
Oh, and `NSCondition`.

Reply:
If the tests pass, I'll amend those too.

I am worried about the failing tests for macOS though.

Reply:
I'll have to defer to @parkera about those failures—I doubt they're related to this PR.

Reply:
Let's go ahead and update `NSRecursiveLock` `NSCondition` and `NSConditionLock`. Thank you for looking into this @maartene !